### PR TITLE
add GET /api/articles/:article:id endpoint

### PR DESCRIPTION
**Pull request by i786m**
ticket 4

add GET /api/articles/:article:id endpoint

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ const db = require("../db/connection");
 const seed = require("../db/seeds/seed");
 const testData = require("../db/data/test-data");
 const endpointsData = require('../endpoints.json')
+const format = require('pg-format');
```
Comment (corkesw) [inline __tests__/app.test.js]: this import looks like it is unused so should be removed

File: __tests__/app.test.js [hunk 2]
```diff
@@ -60,3 +61,43 @@ describe('GET /api', () => {
...
+
+    test('404: should return a 404 error with "Not Found" as the message if passed an id that doesnt exist', () => { 
+        return request(app)
+        .get('/api/articles/55')
+        .expect(404)
+        .then(({ body }) => {
+        expect(body.msg).toBe('Not Found');
+      });
+     })
+})
```
Comment (corkesw) [inline __tests__/app.test.js:100]: Tests look good :)

File: controllers/articles.controllers.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const { selectArticleById } = require( "../models/articles.models" )
+
+exports.getArticlesById = (req,res,next) => {
+ const {article_id} = req.params
+ return selectArticleById(article_id)
+    .then((article)=>{
+     res.status(200).send({article})
+    })
+    .catch(next)
+}
```
Comment (corkesw) [inline controllers/articles.controllers.js:10]: Controller looks good :)

File: models/articles.models.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+exports.selectArticleById = (article_id) => {
+    return db
+    .query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({status : 404 , msg : 'Not Found'})
+      }
+      return rows[0]
+    })
+}
```
Comment (corkesw) [inline models/articles.models.js:12]: Model looks good :)

File: errors/errors.js [hunk 5]
```diff
@@ -1,3 +1,15 @@
...
+  if (err.code === '22P02') {
+    res.status(400).send({ msg: 'Bad Request' });
+  } else next(err);
+};
+
+exports.handleCustomErrors = (err, req, res, next) => {
+  if (err.msg && err.status) {
+    res.status(err.status).send({ msg: err.msg });
+  } else next(err);
+};
```
Comment (corkesw) [inline errors/errors.js:15]: Nice work on the error handlers

**Comment by corkesw** [comment]
Apart from the now unused import, I can't find anything to critique here. Great work. Merge once you have removed the import.

File: __tests__/app.test.js (hunk 1 above)
Comment (i786m) [reply __tests__/app.test.js]: noted and amended

**Pull request merged**